specification/under_development/output_schemas/Labeled_And_Async.md

```diff
@@ -0,0 +1,245 @@
+# Labeled Format for Asynchronous Output Generation
+
+The labeled format for asynchronous output generation is the same as the [ordered format][] with the following changes:
+
+- `OUTPUT` records `TUPLE`, `ARRAY`, `RESULT`, `INT`, `BOOL`, and `DOUBLE` types have a fourth field indicating the label of the record.
+
+A [grammar](./Grammars.md#labeled-and-async) for this format is available which defines the structure and valid values.
+
+Labels are needed for reconstruction of asynchronous/parallel computation output and are assigned by the front-end QIR generator. Order is not important for the `OUTPUT` entries within a `START`/`END` block; however, the responsibility of reconstructing the output based on the defined labeling scheme belongs to the party permforming the output labeling. The usage of `t0_0a` and `t2_2a` (and other values) are examples of a labeling scheme, and are only used as an example.
+
+Consumers of the QIR need to map the associated labels for each recording call to its output entry label. The [base profile required attributes](../profiles/Base_Profile.md#attributes) define the minimum set of attributes which will appear. Examples can be found in the [notes for implementors examples](./Notes_For_Implementors.md#examples).
+
+Example log for a single shot:
+
+```log
+START
+METADATA\tentry_point
+METADATA\tnum_required_qubits\t5
+METADATA\tnum_required_results\t5
```

Comment:
To be consistent with the Base Profile documentation, I believe this should be `required_num_qubits` instead of `num_required_qubits` and `required_num_results` instead of `num_required_results`.